src/core/server/saved_objects/migrations/core/migrate_raw_docs.ts

```diff
@@ -81,26 +81,40 @@ export async function migrateRawDocs(
   return processedDocs;
 }
 
+interface MigrateRawDocsSafelyDeps {
+  serializer: SavedObjectsSerializer;
+  knownTypes: ReadonlySet<string>;
+  migrateDoc: MigrateAndConvertFn;
+  rawDocs: SavedObjectsRawDoc[];
+}
+
 /**
  * Applies the specified migration function to every saved object document provided
  * and converts the saved object to a raw document.
  * Captures the ids and errors from any documents that are not valid saved objects or
  * for which the transformation function failed.
  * @returns {TaskEither.TaskEither<DocumentsTransformFailed, DocumentsTransformSuccess>}
  */
-export function migrateRawDocsSafely(
-  serializer: SavedObjectsSerializer,
-  migrateDoc: MigrateAndConvertFn,
-  rawDocs: SavedObjectsRawDoc[]
-): TaskEither.TaskEither<DocumentsTransformFailed, DocumentsTransformSuccess> {
+export function migrateRawDocsSafely({
+  serializer,
+  knownTypes,
+  migrateDoc,
+  rawDocs,
+}: MigrateRawDocsSafelyDeps): TaskEither.TaskEither<
+  DocumentsTransformFailed,
+  DocumentsTransformSuccess
+> {
   return async () => {
     const migrateDocNonBlocking = transformNonBlocking(migrateDoc);
     const processedDocs: SavedObjectsRawDoc[] = [];
     const transformErrors: TransformErrorObjects[] = [];
     const corruptSavedObjectIds: string[] = [];
     const options = { namespaceTreatment: 'lax' as const };
     for (const raw of rawDocs) {
-      if (serializer.isRawSavedObject(raw, options)) {
+      // Do not transform documents of unknown types
+      if (raw?._source?.type && !knownTypes.has(raw._source.type)) {
+        processedDocs.push(raw);
+      } else if (serializer.isRawSavedObject(raw, options)) {
```

Comment:
Note that this could be done in the `transformDocs` step as well, but seeing as that step is just a proxy to this logic, I thought it was simpler to bake this directly into the same place where all the other processing decisions are being made.

Reply:
The correct place is probably the `DocumentMigrator`. But doing it there is probably fine, and definitely way easier.

Reply:
I didn't want anything that impacted v1 migrations to isolate the change here, so yeah I think this works fine then.